tutorials/R/api-token/neon-api-tokens-tutorial.Rmd

```diff
@@ -0,0 +1,277 @@
+
+```
+
+NEON API tokens are very long, so it would be annoying to keep pasting the 
+entire text string into functions. Assign your token an object name:
+
+```{r nameToken, eval=FALSE, comment=NA}
+
+NEON_TOKEN <- "PASTE YOUR TOKEN HERE"
+
+```
+
+Now we'll use the `loadByProduct()` function to download data. Your 
+API token is entered as the optional `token` input parameter. For 
+this example, we'll download Plant foliar traits (DP1.10026.001).
+
+```{r getCFC, eval=FALSE, comment=NA}
+
+foliar <- loadByProduct(dpID="DP1.10026.001", site="all", 
+                        package="expanded", check.size=F,
+                        token=NEON_TOKEN)
+
+```
+
+You should now have data saved in the `foliar` object; the API 
+silently used your token. If you've downloaded data without a 
+token before, you may notice this is faster!
+
+This format applies to all `neonUtilities` functions that involve 
+downloading data or otherwise accessing the API; you can use the 
+`token` input with all of them. For example, when downloading 
+remote sensing data:
+
+```{r getAOP, eval=FALSE, comment=NA}
+
+chm <- byTileAOP(dpID="DP3.30015.001", site="WREF", 
+                 year=2017, check.size=F,
+                 easting=c(571000,578000), 
+                 northing=c(5079000,5080000), 
+                 savepath=wd,
+                 token=NEON_TOKEN)
+
+```
+
+## Token management for open code
+
+Your API token is unique to your account, so don't share it!
+
+If you're writing code that will be shared with colleagues or available 
+publicly, such as in a GitHub repository or supplemental materials of a 
+published paper, you can't include the line of code above where we assigned 
+your token to `NEON_TOKEN`, since your token is fully visible in the code 
+there. Instead, you'll need to save your token locally on your computer, 
+and pull it into your code without displaying it. There are a few ways to 
+do this, we'll show two options here.
+
+* Option 1: Save the token in a local file, and `source()` that file at the 
+start of every script. This is fairly simple but requires a line of code in 
+every script.
+
+* Option 2: Add the token to a `.Renviron` file to create an environment 
+variable that gets loaded when you open R. This is a little harder to set 
+up initially, but once it's done, it's done globally, and it will work in 
+every script you run.
+
+## Option 1: Save token in a local file
+
+Open a new, empty R script (.R). Put a single line of code in the script:
+
+```{r scriptToSource, eval=FALSE, comment=NA}
+
+NEON_TOKEN <- "PASTE YOUR TOKEN HERE"
+
+```
+
+Save this file in a logical place on your machine, somewhere that won't be 
+visible publicly. Here, let's call the file `neon_token_source.R`, and 
+save it to the working directory. Then, at the start of 
+every script where you're going to use the NEON API, you would run this line 
+of code:
+
+```{r source, eval=FALSE, comment=NA}
+
+source(paste0(wd, "/neon_token_source.R"))
```

Comment:
@cklunch yup looks good! This is also a very cool technique for loading environmental variables that is both simple and effective - great solution!